### ISSUE TITLE
Hopper build fails on Windows 10

### ISSUE THREAD

**Issue by symexec** [FuzzAnything/Hopper#16]
I am trying to build Hopper (Commit: `e73a03a`) on Windows 10. My build environment is as follows:

```
OS: Windows 10.0.19044.1288
Compiler: rustc 1.75.0 (82e1608df 2023-12-21)
Toolchain: stable-x86_64-pc-windows-gnu
Backend: gcc 11.4.0 (Cygwin)
```

I am hitting build issues in the platform-specific code. I tried to hand-fix some of them, but quickly I hit my limit. Has this version (the latest one at the time of posting this issue) of Hopper been tested on Windows 10 (or any other version of Windows)? If you did, can you please share the details of the build environment?

**Comment by spinpx** [comment]
Hi @symexec, sorry that I am on holiday these days. Hopper's Windows support is experimental and not maintained (as mentioned in https://github.com/FuzzAnything/Hopper/blob/main/README_Windows.md), and I do not have a Windows environment. This work was done by @AiDaiP.

@AiDaiP, could you help @symexec fix this issue?

**Comment by AiDaiP** [comment]
Hi @symexec, sorry for not replying in time.
The latest version of Hopper has not been tested on Windows. Compared with the version I implemented for windows support, hopper has added many new features, and the code has also changed a lot. Code related to Windows has not been updated in time, so some compilation errors will occur (such as import errors, type errors, etc.). In addition, some new features cannot be compiled directly on Windows. These new features need to be implemented on Windows.
I tried to modify some simple compilation errors and marked some code that could not be compiled with `#[cfg(target_os = "linux")]` and ignored them. After modification, it can be compiled successfully and generate harness, but I cannot guarantee that this is correct because problems may occur at runtime.
